### ISSUE TITLE
BLT Sync Commands Fail

### ISSUE THREAD

**Issue by mikemadison13** [acquia/blt#2075]
My system information:
* Operating system type: Mac OS
* Operating system version: 10.11.6
* BLT version: 8.9.x-dev

Output of `blt doctor`:
```
blt doctor
Drupal VM was detected. Running blt doctor inside of VM...
Executing command cd /var/www/<project>/docroot/.. && /var/www/<project>/docroot/../vendor/bin/drush cc drush && /var/www/<project>/docroot/../vendor/bin/drush --include=/var/www/<project>/docroot/../vendor/acquia/blt/drush blt-doctor -r /var/www/<project>/docroot/../docroot inside of Drupal VM...
 [success] 'drush' cache was cleared.
 [notice] Behat base_url matches drush URI.
+-----------------------------+----------------------------------------------------------------------------------+
| Property                    | Value                                                                            |
+-----------------------------+----------------------------------------------------------------------------------+
| %paths                      |                                                                                  |
|  - %root                    | /var/www/<project>/docroot/../docroot                                                |
|  - %site                    | sites/default                                                                    |
|  - %modules                 | sites/all/modules                                                                |
|  - %themes                  | sites/all/themes                                                                 |
|  - %config-sync             | ../config/default                                                                |
|  - %files                   | sites/default/files                                                              |
|  - %temp                    | /tmp                                                                             |
|  - %private                 | /var/www/<project>/files-private                                                     |
| admin-theme                 | seven                                                                            |
| blt-version                 | 8.9.x-dev                                                                        |
| bootstrap                   | Successful                                                                       |
| config-sync                 | ../config/default                                                                |
| db-driver                   | mysql                                                                            |
| db-hostname                 | localhost                                                                        |
| db-name                     | drupal                                                                           |
| db-password                 | drupal                                                                           |
| db-port                     | 3306                                                                             |
| db-status                   | Connected                                                                        |
| db-username                 | drupal                                                                           |
| dev-desktop-enabled         | false                                                                            |
| drupal-settings-file        | sites/default/settings.php                                                       |
| drupal-version              | 8.4.0-dev                                                                        |
| drupal-vm-enabled           | true                                                                             |
| drush-alias-files           | /var/www/<project>/docroot/../docroot/../drush/site-aliases/aliases.drushrc.php      |
|                             | /var/www/<project>/docroot/../docroot/../drush/site-aliases/example.acsf.aliases.dru |
|                             | shrc.php                                                                         |
|                             | /var/www/<project>/docroot/../docroot/../drush/site-aliases/example.local.aliases.dr |
|                             | ushrc.php                                                                        |
|                             | /var/www/<project>/docroot/../docroot/../drush/site-aliases/<project>.aliases.drushrc.ph |
|                             | p                                                                                |
| drush-conf                  | /var/www/<project>/drush/drushrc.php                                                 |
| drush-script                | /var/www/<project>/vendor/bin/drush                                                  |
| drush-temp                  | /tmp                                                                             |
| drush-version               | 9.0.0-beta4                                                                      |
| files                       | sites/default/files                                                              |
| install-profile             | lightning                                                                        |
| local-drushrc               | /var/www/<project>/docroot/../docroot/sites/default/local.drushrc.php                |
| local-settings              | /var/www/<project>/docroot/../docroot/sites/default/settings/local.settings.php      |
| modules                     | sites/all/modules                                                                |
| php-bin                     | /usr/bin/php7.1                                                                  |
| php-conf                    |                                                                                  |
|  - /etc/php/7.1/cli/php.ini | /etc/php/7.1/cli/php.ini                                                         |
| php-mysql                   | /var/run/mysqld/mysqld.sock                                                      |
| php-os                      | Linux                                                                            |
| private                     | /var/www/<project>/files-private                                                     |
| root                        | /var/www/<project>/docroot/../docroot                                                |
| shell                       |                                                                                  |
| site                        | sites/default                                                                    |
| temp                        | /tmp                                                                             |
| theme                       | bartik                                                                           |
| themes                      | sites/all/themes                                                                 |
| uri                         | http://local.<project>.com                                                           |
+-----------------------------+----------------------------------------------------------------------------------+
+--------------------------+----------------------------------------------------------------------------------+
| Check                    | Outcome                                                                          |
+--------------------------+----------------------------------------------------------------------------------+
| checkDocrootExists       | Found docroot.                                                                   |
| checkCoreExists          | Drupal core exists                                                               |
| checkSettingsFile        | BLT settings are included in settings file.                                      |
| checkLocalSettingsFile   | Found your local settings file.                                                  |
| checkLocalDrushFile      | Found your local drush settings file.                                            |
| checkUri                 | $options['uri'] is set.                                                          |
| checkUriResponse         | Received a response from site http://local.<project>.com.                            |
| checkFileSystem:%files   | Public files directory is writable.                                              |
| checkFileSystem:%private | Private files directory is writable.                                             |
| checkFileSystem:%temp    | Temporary files directory is writable.                                           |
| checkDbConnection        | Connected to database.                                                           |
| checkDrupalBootstrapped  | Bootstrapped Drupal via drush.                                                   |
| checkDrupalInstalled     | Drupal is installed.                                                             |
| checkCaching:page        | Drupal cache is disabled.                                                        |
| checkCaching:css         | CSS preprocessing is disabled.                                                   |
| checkCaching:js          | JS preprocessing is disabled.                                                    |
| checkCiConfig            | Git remotes are set in project.yml.                                              |
| checkComposer:require    | acquia/blt is in composer.json's require object.                                 |
| checkBehat:exists        | Behat local settings file exists.                                                |
| checkBehat:root          | Behat drupal_root is set correctly for Drupal VM.                                |
| checkProjectYml:keys     | project.yml has no deprecated keys.                                              |
| checkAcsfConfig          | BLT settings are not included in your pre-settings-php include.                  |
|                          |   Add a require statement for "/../vendor/acquia/blt/settings/blt.settings.php"  |
|                          | to                                                                               |
|                          | /var/www/<project>/docroot/../docroot/../factory-hooks/pre-settings-php/includes.php |
| checkDrupalVm:alias      | drush.aliases.local exists your drush aliases file.                              |
| checkDrupalVm            | Drupal VM is configured correctly.                                               |
| checkPhpDateTimezone     | PHP setting for date.timezone is correctly set                                   |
+--------------------------+----------------------------------------------------------------------------------+
```

When I run this command:
```
blt sync
```

I get the following output:
```
blt sync -vvv
[Robo\Common\ProcessExecutor] Running vagrant status --machine-readable in /Users/mike.madison/git/<project>
[debug] vagrant <project>.metadata = provider
[debug] vagrant <project>.provider-name = virtualbox
[debug] vagrant <project>.state = running
[debug] vagrant <project>.state-human-short = running
[debug] vagrant <project>.state-human-long = The VM is running. To stop this VM%!(VAGRANT_COMMA) you can run `vagrant halt` to\nshut it down forcefully%!(VAGRANT_COMMA) or you can run `vagrant suspend` to simply\nsuspend the virtual machine. In either case%!(VAGRANT_COMMA) to restart it again%!(VAGRANT_COMMA)\nsimply run `vagrant up`.
[debug] vagrant .ui = info
[debug] Drupal VM is initialized.
[debug] Drupal VM is booted.
> sync:db
> setup:settings
[Filesystem\FilesystemStack] _chmod ["/Users/mike.madison/git/<project>/docroot/sites/default",511]
[Filesystem\FilesystemStack] _chmod ["/Users/mike.madison/git/<project>/docroot/sites/default/settings.php",511]
[File\Write] Writing to /Users/mike.madison/git/<project>/docroot/sites/default/settings.php.
[Filesystem\FilesystemStack] _chmod ["/Users/mike.madison/git/<project>/docroot/sites/default/settings.php",420]
[Acquia\Blt\Robo\Tasks\DrushTask] Running /Users/mike.madison/git/<project>/vendor/bin/drush cache-clear drush --uri=default --yes && /Users/mike.madison/git/<project>/vendor/bin/drush sql-drop --uri=default --yes && /Users/mike.madison/git/<project>/vendor/bin/drush sql-sync @mtav2.test @<project>.local --structure-tables-key=lightweight --create-db --sanitize --uri=default --yes && /Users/mike.madison/git/<project>/vendor/bin/drush cache-clear drush --uri=default --yes in /Users/mike.madison/git/<project>/docroot
 [success] 'drush' cache was cleared.

 // Do you really want to drop all tables in the database drupal?: yes.                                                 

 [error]  Unable to drop database. Rerun with --debug to see any error message. 
[Acquia\Blt\Robo\Tasks\DrushTask]  Exit code 1  Time 0.645s
[error]   
[error]  Command `sync:db ` exited with code 1. 

```

And I expected this to happen:

**Comment by grasmash** [comment]
This will be resolved in #2070.

**Comment by mikemadison13** [comment]
that commit gets sync running again. thanks!
